src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.servicebus.MessageBody;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.exceptions.InvalidMessageException;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.model.PaymentMessage;
+
+import java.io.IOException;
+import java.util.List;
+
+@Service
+public class PaymentMessageParser {
+
+    @Autowired
+    private final ObjectMapper objectMapper;
+
+    private static final Throwable errorCause = new RuntimeException("Message Binary data is null");
```

Comment:
??

Reply:
to throw exception as InvalidMessageException wraps Throwable.

Reply:
removed Throwable, return message